examples/fluids/qfunctions/newtonian.h

```diff
@@ -93,6 +93,40 @@ CEED_QFUNCTION(ICsNewtonianIG_Prim)(void *ctx, CeedInt Q, const CeedScalar *cons
   return 0;
 }
 
+// *****************************************************************************
+// This QFunction returns the state variables from conservative variables to
+//  primitive variables or vice versa.
+// *****************************************************************************
+CEED_QFUNCTION(NewtonianConvertState)(void *ctx, CeedInt Q, const CeedScalar *const *in, CeedScalar *const *out) {
+  // Inputs
+  const CeedScalar(*X)[CEED_Q_VLA]        = (const CeedScalar(*)[CEED_Q_VLA])in[0];
+  const CeedScalar(*q_source)[CEED_Q_VLA] = (const CeedScalar(*)[CEED_Q_VLA])in[1];
+
+  // Outputs
+  CeedScalar(*q_target)[CEED_Q_VLA] = (CeedScalar(*)[CEED_Q_VLA])out[0];
+
+  NewtonianIdealGasContext context = (NewtonianIdealGasContext)ctx;
+  // Quadrature Point Loop
+  CeedPragmaSIMD for (CeedInt i = 0; i < Q; i++) {
+    const CeedScalar x[3]   = {X[0][i], X[1][i], X[2][i]};
+    CeedScalar       q_s[5] = {0}, q_t[5] = {0};
+    for (CeedInt j = 0; j < 5; j++) q_s[j] = q_source[j][i];
+    State s;
+    switch (context->state_var) {
+      case STATEVAR_CONSERVATIVE:
+        s = StateFromU(context, q_s, x);
+        StateToY(context, s, q_t);
+        break;
+      case STATEVAR_PRIMITIVE:
+        s = StateFromY(context, q_s, x);
+        StateToU(context, s, q_t);
+        break;
+    }
```

Comment:
How are we going to maintain this once entropy variables (#1110) merge? Should it instead be converting from whatever to primitive?

Reply:
Oh, this was my concern too and tried to create a new restriction to be able to output 10 values instead of 5. However, I didn't find it easy, so gave up due to the timing of the course project. I believe we would need to report the error in all state variables if we want to perform an analysis similar to what we did with the traveling vortex problem in summer. That said, I don't really know how to do it without making the algorithm too complicated.

Reply:
What's wrong with having `ToPrimitive` and perhaps `ToConservative`, where either can be NULL. Entropy variables can implement both. In practice, I think >90% of comparisons will be comparing primitive variables because they are more intuitive and interpretable, so I think you could just have `ToPrimitive` and call it a day. If someone needs to compare conservative variables directly, they can add that alternative when the time comes.

Reply:
I guess I was worried about having too many QFunctions and if-statements. Alright, I'll keep `ToPrimitive` for now and output the errors for Euler and Advection2D by using a second flag.